src/software/amazon/ion/impl/IonReaderBinaryRawX.java

```diff
@@ -364,6 +364,9 @@ private final IonType load_annotation_start_with_value_type() throws IOException
         if (_value_tid == UnifiedInputStreamX.EOF) {
             throwErrorAt("unexpected EOF encountered where a type descriptor byte was expected");
         }
+        if (_value_tid == PrivateIonConstants.tidTypedecl) {
+            throwErrorAt("Malformed ION, annotation of an annotation found.");
```

Comment:
I'd suggest a message like:
"An annotation wrapper may not contain another annotation wrapper."

We know it's malformed Ion because an error is being thrown, and "annotation of an annotation" is a bit ambiguous.